purity/gc/gc.go

```diff
@@ -492,24 +493,28 @@ func (c *collector) RemoveActionResults() error {
 	wg.Add(c.parallelism + 1)
 	step := numArs / c.parallelism
 	for i := 0; i < (c.parallelism + 1); i++ {
+		i := i
```

Comment:
This is interesting, why do we have to assign i to i. Is it a concurrency thing?

Reply:
It's a for loop thing in go. https://github.com/golang/go/discussions/56010 This covers what they want to change about the semantics!

Reply:
Although, I think `i` isn't used, so there's no need I think?

Reply:
That's correct. The nightly job occasionally fails early with no warning and I assumed it may have been an issue here. Probably due to not locking when writing to a map below though